src/Controls/tests/Core.UnitTests/BindableLayoutTests.cs

```diff
@@ -530,5 +601,72 @@ public MyDataTemplateSelectorTest(Func<object, BindableObject, DataTemplate> fun
 			protected override DataTemplate OnSelectTemplate(object item, BindableObject container)
 				=> _func(item, container);
 		}
+		
+		class MyViewModel : INotifyPropertyChanged
+		{
+			string _text;
+
+			public event PropertyChangedEventHandler PropertyChanged;
+
+			public string Text
+			{
+				get => _text;
+				set
+				{
+					_text = value;
+					PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Text)));
+				}
+			}
+		}
+
+		/// <summary>
+		/// When binding context is changed, the component watches for changes in the view model `Text` property.
+		/// The event handler is removed correctly when the binding context is changed or cleared.
+		/// </summary>
+		class MyViewModelBoundComponent : ContentView
+		{
+			public static int InstanceCount = 0;
+			
+			readonly Action _onTextChangedCallback;
+			MyViewModel _myViewModel;
+			
+			public string Text => _myViewModel?.Text;
+			
+			public MyViewModelBoundComponent(Action onTextChangedCallback)
+			{
+				++InstanceCount;
+				_onTextChangedCallback = onTextChangedCallback;
+			}
+			
+			~MyViewModelBoundComponent()
+			{
+				--InstanceCount;
+			}
+			
+			protected override void OnBindingContextChanged()
+			{
+				base.OnBindingContextChanged();
+
+				if (_myViewModel != null)
+				{
+					_myViewModel.PropertyChanged -= OnBindingContextFixturePropertyChanged;
+				}
+
+				_myViewModel = BindingContext as MyViewModel;
+
+				if (_myViewModel != null)
+				{
+					_myViewModel.PropertyChanged += OnBindingContextFixturePropertyChanged;
+				}
```

Comment:
Is this line the actual cause of the leak? You've created a control and viewmodel where the viewmodel has a strong reference to the control. Then at the top of the test you have a viewmodel that lives the lifetime of the test:

```csharp
var myViewModel = new MyViewModel();
```
You would need to enclose `myViewModel` in a scope, so the GC can collect it?

I'm still not sure there is an actual memory leak here, besides the one created in this test. Clearing the `BindingContext` just makes the event unsubscribe in this custom control and remove the strong reference. I would consider not designing a custom control this way if the viewmodel is known to live longer than the control.

Maybe someone on the MAUI team can comment what the behavior is supposed to be in regards to `BindingContext`. Should it always get cleared when children are unparented? @PureWeen @mattleibow @jsuarezruiz?

Reply:
I think the point here is:
What should a View do if it needs to subscribe on a BindingContext property to do some action when something changes there?
How can it unsubscribe properly in the exact moment the view is no more needed?
The view cannot rely on the GC because that might happen at later moment in time, and something can happen in the mean time which would trigger an action on a removed view.

Reply:
It might be correct to clear the `BindingContext` (MAUI team can comment), but these tests have simply created a control that leaks unless its `BindingContext` is cleared. It might be better to use `WeakEventManager` or other solutions to avoid the problem entirely.

Does that make sense?

Reply:
It might make sense. Is the WeakEventManager instantly reacting to the un-reference of the child view?

Anyway for example, `ObservableObject` in community toolkit does not use weak event manager.

https://github.com/CommunityToolkit/dotnet/blob/main/src/CommunityToolkit.Mvvm/ComponentModel/ObservableObject.cs#L34

I'm just saying this kind of situations happen in complex applications, and we need a clear way to handle them.
I felt that relaying on the way binding context inheritance works was the way.